src/compiler/scanner.ts

```diff
@@ -2116,19 +2149,19 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
                         }
                     }
 
-                    if (text.charCodeAt(pos + 1) === CharacterCodes.lessThan) {
-                        if (text.charCodeAt(pos + 2) === CharacterCodes.equals) {
+                    if (charCodeUnchecked(pos + 1) === CharacterCodes.lessThan) {
+                        if (charCodeUnchecked(pos + 2) === CharacterCodes.equals) {

```

Comment:
Many of these should probably be switched over to the checked version, but just never trigger because it's rare to hit these tokens at the end of a file, right?

Reply:
See #58362 for most of the rest of scanner.